spec/factories/evaluation.rb

```diff
@@ -11,6 +11,20 @@
       terminee_le { Time.current }
     end
 
+    trait :avec_donnee_sociodemographique do
+      transient do
+        age { nil }
+        derniere_situation { nil }
+      end
+      after(:create) do |evaluation, evaluator|
+        create(:donnee_sociodemographique, evaluation: evaluation)
+        evaluation.donnee_sociodemographique.update(
+          age: evaluator.age,
+          derniere_situation: evaluator.derniere_situation
+        )
+      end
+    end
```

Comment:
Je n'ai pas l'impression que tu utilises la fonctionnalité de transient, qui permet de passer de la data pour des attributs autre que celui du modèle. Tu aurais juste pu utiliser la factory existante de donne_sociodemographique et l'ajouter au trait il me semble`:avec_donnee_sociodemographique`

```suggestion
    trait :avec_donnee_sociodemographique do
      after(:create) do |evaluation|
        create(:donnee_sociodemographique, evaluation: evaluation)
      end
    end
```